sklearn/metrics/cluster/tests/test_common.py

```diff
@@ -0,0 +1,171 @@
+SUPERVISED_METRICS = {
+    "adjusted_mutual_info_score": adjusted_mutual_info_score,
+    "adjusted_rand_score": adjusted_rand_score,
+    "completeness_score": completeness_score,
+    "homogeneity_score": homogeneity_score,
+    "mutual_info_score": mutual_info_score,
+    "normalized_mutual_info_score": normalized_mutual_info_score,
+    "v_measure_score": v_measure_score,
+    "fowlkes_mallows_score": fowlkes_mallows_score
+}
+
+UNSUPERVISED_METRICS = {
+    "silhouette_score": silhouette_score,
+    "silhouette_manhattan": partial(silhouette_score, metric='manhattan'),
+    "calinski_harabaz_score": calinski_harabaz_score
+}
+
+# Lists of metrics with common properties
+# ---------------------------------------
+# Lists of metrics with common properties are used to test systematically some
+# functionalities and invariance, e.g. SYMMETRIC_METRICS lists all metrics
+# that are symmetric with respect to their input argument y_true and y_pred.
+#
+# --------------------------------------------------------------------
+# Symmetric with respect to their input arguments y_true and y_pred.
+# Symmetric metrics only apply to supervised clusters.
+SYMMETRIC_METRICS = [
+    "adjusted_rand_score", "v_measure_score",
+    "mutual_info_score", "adjusted_mutual_info_score",
+    "normalized_mutual_info_score", "fowlkes_mallows_score"
+]
+
+ASYMMETRIC_METRICS = ["homogeneity_score", "completeness_score"]
+
+
+# Metrics whose upper bound is 1
+METRICS_NORMALIZED_OUTPUT = [
+    "adjusted_rand_score", "homogeneity_score", "completeness_score",
+    "v_measure_score", "adjusted_mutual_info_score", "fowlkes_mallows_score",
+    "normalized_mutual_info_score"
+]
+
+
+def assert_between(var, score_1, score_2):
+    """Helper function to check if score lies in between two values"""
+    if assert_greater(var, score_1) and assert_less(var, score_2):
+        raise AssertionError("score does not lie between 0 and 1")
+
+
+def test_symmetry():
+    rng = np.random.RandomState(0)
+    y1 = rng.randint(3, size=30)
+    y2 = rng.randint(3, size=30)
+    for name in SYMMETRIC_METRICS:
+        metric = SUPERVISED_METRICS[name]
+        assert_almost_equal(metric(y1, y2), metric(y2, y1))
+
+    for name in ASYMMETRIC_METRICS:
+        metric = SUPERVISED_METRICS[name]
+        assert_not_equal(metric(y1, y2), metric(y2, y1))
+
+
+def test_normalized_output():
+    upper_bound_1 = [0, 0, 0, 1, 1, 1]
+    upper_bound_2 = [0, 0, 0, 1, 1, 1]
+    for name in METRICS_NORMALIZED_OUTPUT:
+        metric = SUPERVISED_METRICS[name]
+        assert_between(metric([0, 0, 0, 1, 1], [0, 0, 0, 1, 2]), 0.0, 1.0)
+        assert_between(metric([0, 0, 1, 1, 2], [0, 0, 1, 1, 1]), 0.0, 1.0)
+        assert_equal(metric(upper_bound_1, upper_bound_2), 1.0)
+
+    lower_bound_1 = [0, 0, 0, 0, 0, 0]
+    lower_bound_2 = [0, 1, 2, 3, 4, 5]
+    for name in SUPERVISED_METRICS:
+        metric = SUPERVISED_METRICS[name]
+        assert_equal((metric(lower_bound_1, lower_bound_2) and

```

Comment:
Yes, I have combined both symmetric and non-symmetric metrics . That,s why I used and .